docs/source/check_docs_api_coverage.py

```diff
@@ -68,12 +70,33 @@
 
             for method in methods:
                 method = ".".join([class_name, method])
-                if method not in rst_contents:
+                count = rst_contents.count(method)
+                if count == 0:
                     n_undocumented_methods += 1
                     print(
                         f"Method {method} not in "
                         f"{os.path.join(source, 'class', rst_file)}"
                     )
+                elif count > 1:
+                    # The method appears more than once, but may be a
+                    # sub-string of another method name, e.g. this gets caught:
+                    #     [cfdm.List.]nc_set_variable
+                    # due to the presence of this method:
+                    #     [cfdm.List.]nc_set_variable_groups
+                    # so we must account for that. Checking next character
+                    # of duplicate(s) is something other than a newline or
+                    # whitespace, seems robust and simplest.
+                    end_loc = [
```

Comment:
I presume I'm missing something here, but `count` doesn't count partial matches, so why is this needed?
```python
>>> a = ['abc', 'abcd', 'abcde']
>>> a.count('abc')
1

Reply:
I think it's needed because what's being done is count from one larger string, being the string read from the full file, rather than a list of sub-strings as per your example, in which case we have as an equivalent:

```python
>>> a = "abc abcd abcde"
>>> a.count("abc")
3

```

Reply:
Regardless, if you can think of a more elegant way to go about the logic I've added, I am happy to hear. I might be able to think of an alternative myself if I mull over it, but since it's a convenience script for automating checks I am satisfied as long as it works and isn't (too) slow :)

Reply:
All good - thanks for the explanation